src/analyzer/engines/gensim.py

```diff
@@ -0,0 +1,71 @@
+from src.analyzer.engines.engine import ModelingEngine
+
+from gensim.models.coherencemodel import CoherenceModel
+from gensim.models.ldamodel import LdaModel
+from gensim.models.tfidfmodel import TfidfModel
+from gensim.models import Nmf
+
+class GensimEngine(ModelingEngine):
+    """
+    GensimEngine implements the methods for fitting and evaluating gensim models on a
+    corpus.
+    """
+    def __init__(self, min_topics, max_topics):
+        self.min_topics = min_topics
+        self.max_topics = max_topics
+        self.models = {}
+        self.dictionary = None
+        self.type = ""
+        self.corpus = None
```

Comment:
Perhaps we could use the sklearn style of postfixing a trailing underscore to any attributes that will be [created by the fit process](https://scikit-learn.org/stable/developers/develop.html#estimated-attributes)? e.g. `self.models_`, `self.dictionary_`, etc.